src/components/PostDetails.tsx

```diff
@@ -4,103 +1,76 @@
+import React, { useEffect, useState } from 'react';
+
+import { Loader, CommentItem, NewCommentForm } from './index';
+import { Post } from '../types';
+import { useComments } from '../hooks/useComments';
+
+type Props = {
+  selectedPost: Post;
+};
+
+export const PostDetails: React.FC<Props> = ({ selectedPost }) => {
+  const { id, title, body } = selectedPost;
+
+  const [isFormOpened, setisFormOpened] = useState(false);
+  const { comments, errorMessage, isLoading, deleteComment, addComment } =
+    useComments(id);
+
+  useEffect(() => setisFormOpened(false), [selectedPost]);
 
-export const PostDetails: React.FC = () => {
   return (
     <div className="content" data-cy="PostDetails">
       <div className="content" data-cy="PostDetails">
```

Comment:
The `data-cy="PostDetails"` attribute is duplicated. Ensure that this is intentional, or remove the duplicate to avoid confusion.